projects/core/components/root/root.component.ts

```diff
@@ -26,10 +27,18 @@ import {map} from 'rxjs/operators';
     },
 })
 export class TuiRootComponent {
-    readonly scrollbars$: Observable<boolean> =
-        this.dialogs.length && !this.isMobile
-            ? merge(...this.dialogs).pipe(map(({length}) => !length))
-            : of(!this.isMobile);
+    @tuiPure
+    get isScrollbarShown$(): Observable<boolean> {
```

Comment:
I can't agree that it's better

What do you think about make a custom pipe?

```html
<tui-scroll-controls
  *ngIf="dialogs | tuiScrollbars | async"
```

```ts
@Pipe({ name: 'tuiScrollbars' })
export class TuiScrollbarPipe implements PipeTransform {
   constructor(
       @Inject(DOCUMENT) {body}: Document,
       @Inject(TUI_SCROLLBAR_MODE)
       private readonly tuiScrollbar: TuiScrollbarMode,,
       @Inject(TUI_IS_MOBILE) private readonly isMobile: boolean,
    ) {
       // ps: not sure mutate body is good for disable scrollbar
        const cssClass = 'tui-zero-scrollbar';
        this.tuiScrollbar.enable
            ? body.classList.remove(cssClass)
            : body.classList.add(cssClass);
   }

   transform<T>(dialogs: ReadonlyArray<Observable<readonly T[]>>): Observable<boolean> {
      if (this.isMobile || !this.tuiScrollbar.enable) {
            return of(false);
        }

        if (!this.dialogs.length) {
            return of(true);
        }

        return merge(...this.dialogs).pipe(map(({length}) => !length));
   }
}
```

Reply:
> not sure mutate body is good for disable scrollbar

any ideas? I tried multiple but thought it was the best.

1. I couldn’t apply the mixin in main.less depending on children state
2. I tried to control it in tui-root + ng deep html, but it looked like a hack

Reply:
it’s not clear to me why we name it ‘tuiScrollbars’ while it returns a boolean. It sounds like it should return an array of scrollbars 🤔

Reply:
`it’s not clear to me why we name it ‘tuiScrollbars’`

you can suggest any solution, it was just an example

`any ideas? I tried multiple but thought it was the best.`

you can leave it like that, maybe `@waterplea` offer a better solution